pkg/pool-utils/contracts/protocol-fees/ProtocolFees.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+library ProtocolFees {
+    using FixedPoint for uint256;
+
+    /**
+     * @dev Calculates the amount of BPT necessary to give ownership of a given percentage of the Pool.
+     * Note that this function reverts if `poolPercentage` >= 100%, it's expected that the caller will enforce this.
+     * @return bptAmount - The amount of BPT to mint such that it is `poolPercentage` of the resultant total supply.
+     */
+    function bptForPoolPercentage(uint256 totalSupply, uint256 poolPercentage) internal pure returns (uint256) {
```

Comment:
Should this be called something like bptForProtocolOwnershipPercentage? I'm afraid it's confusable with the fee percentage itself, which it *might* correspond to (e.g., for managed AUM fees), but not always, when we're really adjusting a pre-mint to a post-mint percentage of ownership.

Maybe, to be totally clear:

function bptForProtocolOwnershipPercentage(uint256 totalSupply, uint256 preMintProtocolOwnershipPercentage)

Reply:
I'd like to avoid repeating "protocol" as it's included in the library name (also as we'll be using this for manager fees so it's not entirely true). I've changed it to:

`function bptForPoolOwnershipPercentage(uint256 totalSupply, uint256 poolOwnershipPercentage)`

>  we're really adjusting a pre-mint to a post-mint percentage of ownership.

Not sure what you mean by this. The percentage ownership should be the same pre and post mint (just beforehand the protocol's ownership is through debt rather than BPT).